backend/models/teamspaceSetting.js

```diff
@@ -101,15 +101,16 @@ class TeamspaceSettings {
 				return Promise.reject(responseCodes.FILE_FORMAT_NOT_SUPPORTED);
 			}
 
+			const Mitigation = require("./mitigation");
+			const readCSVProm = await Mitigation.importCSV(account, file);
+
 			const storeFileProm = FileRef.storeMitigationsFile(account, username, filename, file).then(async () => {
 				const settingsCol = await this.getTeamspaceSettingsCollection(account, true);
 				const updatedAt = new Date();
 				await settingsCol.update({_id: account}, {$set: {"mitigationsUpdatedAt":updatedAt}});
 				return updatedAt;
 			});
 
-			const Mitigation = require("./mitigation");
-			const readCSVProm = Mitigation.importCSV(account, file);
 			return Promise.all([storeFileProm, readCSVProm]);
```

Comment:
We're awaiting readCSVProm higher up, so i don't think readCSVProm is really a promise anymore?